### PR TITLE
TST: Bare pytest raises

### PR DESCRIPTION
**Pull request by MJafarMashhadi**
- xref #30999

- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry [N/A]

### CODE REVIEW

File: pandas/tests/extension/base/ops.py [hunk 1]
```diff
@@ -30,7 +30,17 @@ def _check_op(self, s, op, other, op_name, exc=NotImplementedError):
...
+                "(unsupported operand type\\(s\\) for)|"
+                "(cannot perform [\\w_]+ with this index type: [\\w_]+)|"
+                "(Object with dtype category cannot perform the numpy op [\\w_]+)|"
+                "(cannot add [\\w_]+ and [\\w_]+)|"
+                "(can't multiply sequence by non-int of type '[\\w_]+')|"
+                '(can only concatenate str \\(not "[\\w_]+"\\) to str)|'
+                "(Object with dtype category cannot perform the numpy op [\\w_]+)|"
+                "(Concatenation operation is not implemented for NumPy arrays)"
+            )
+            with pytest.raises(exc, match=msg):
```
Comment (MarcoGorelli) [inline pandas/tests/extension/base/ops.py:43]: nice! I haven't tried running this, but I would have thought that the parentheses would be redundant - does
```suggestion
            msg = (
                "unsupported operand type\\(s\\) for|"
                "cannot perform [\\w_]+ with this index type: [\\w_]+|"
                "Object with dtype category cannot perform the numpy op [\\w_]+|"
                "cannot add [\\w_]+ and [\\w_]+|"
                "can't multiply sequence by non-int of type '[\\w_]+'|"
                'can only concatenate str \\(not "[\\w_]+"\\) to str|'
                "Object with dtype category cannot perform the numpy op [\\w_]+|"
                "Concatenation operation is not implemented for NumPy arrays"
            )
            with pytest.raises(exc, match=msg):
```
work?

**Review by MarcoGorelli — APPROVED** [review]
Looks good to me

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @MJafarMashhadi

**Comment by jorisvandenbossche** [comment]
This are base extension tests, which are also used by external projects. IMO, we should not be adding such specific message tests. I would propose to revert this change. 

(and @MJafarMashhadi, to be clear, not your fault!)

**Comment by MarcoGorelli** [comment]
Makes sense

Should this be reverted and then ` ^pandas/tests/extension/` excluded from when the "disallow bare pytest raises" will be applied to CI, or should something generic like `match=''` be added here?

**Comment by xhochy** [comment]
Yes, please revert that. I need to hardcode specific messages now in https://github.com/xhochy/fletcher/pull/212

**Comment by jorisvandenbossche** [comment]
Will do a revert tomorrow

**Comment by MJafarMashhadi** [comment]
@jorisvandenbossche There are 4 files that are changed in this PR, two of them are not in `extensions/` should those be reverted too?

**Comment by jorisvandenbossche** [comment]
The non-extension tests in `test_assert_series_equal.py` were not reverted in #38748

**Pull request merged**